src/requireLocation.ts

```diff
@@ -0,0 +1,10 @@
+export const requireLocation = (store: any) => {
+  return (nextState: any, replace: any) => {
+    if (!store.getState().user.location) {
+      replace({
+        pathname: '/',
+        state: { nextPathname: nextState.location.pathname }
```

Comment:
is this so that we can continue the user on to their attempted page after completing popup?